shell/run_shell_cmd.go

```diff
@@ -268,23 +269,31 @@ type CmdOutput struct {
 	Stderr string
 }
 
+var gitTopLevelDirs = cache.NewGenericCache[string]()
+
 // GitTopLevelDir - fetch git repository path from passed directory
 func GitTopLevelDir(terragruntOptions *options.TerragruntOptions, path string) (string, error) {
-	stdout := bytes.Buffer{}
-	stderr := bytes.Buffer{}
-	opts, err := options.NewTerragruntOptionsWithConfigPath(path)
-	if err != nil {
-		return "", err
-	}
-	opts.Env = terragruntOptions.Env
-	opts.Writer = &stdout
-	opts.ErrWriter = &stderr
-	cmd, err := RunShellCommandWithOutput(opts, path, true, false, "git", "rev-parse", "--show-toplevel")
-	terragruntOptions.Logger.Debugf("git show-toplevel result: \n%v\n%v\n", (string)(stdout.Bytes()), (string)(stderr.Bytes()))
-	if err != nil {
-		return "", err
+	gitTopLevelDir, found := gitTopLevelDirs.Get(path)
+	if !found {
```

Comment:
Nested blocks can be reduced if execution flow will exit if value is cached
```
	if gitTopLevelDir, found := gitTopLevelDirs.Load(path); found {
		return gitTopLevelDir.(string), nil
	}

...
```

Reply:
Yes, that is indeed more elegant, I corrected it, thanks